src/tdastro/astro_utils/dustmap.py

```diff
@@ -144,3 +145,50 @@ def query(self, coords):
             The E(B-V) value or array of values.
         """
         return self.compute_ebv(coords.ra.deg, coords.dec.deg)
+
+
+class SFDMap(DustEBV):
+    """
+    A dustmap using the sfdmap2 package (https://github.com/AmpelAstro/sfdmap2)
+    """
+
+    def __init__(self, **kwargs):
+        super().__init__(self.compute_ebv, **kwargs)
+
+    def compute_ebv(self, ra, dec):
+        """Compute the E(B-V) value for a given location.
+
+        Parameters
+        ----------
+        ra : float or np.array
+            The object's right ascension (in degrees).
+        dec : float or np.array
+            The object's declination (in degrees).
+
+        Returns
+        -------
+        ebv : float or np.array
+            The E(B-V) value or array of values.
+        """
+
+        dustmap = sfdmap.SFDMap()
```

Comment:
This should go in the constructor. Something like:

```
self.data_dir = self. _default_map_dir is data_dir is None else data_dir
self.dustmap = sfdmap.SFDMap(self.data_dir)
```

That way it is only loaded once.